bin/summarize-user-events

```diff
@@ -57,6 +55,32 @@ class SummarizeUserEvents
       log_group_name: 'prod_/srv/idp/shared/log/events.log',
     )
   end
+
+  def find_cloudwatch_events(&block)
+    if $stdin.tty?
+      cloudwatch_source(&block)
+    else
+      warn "Reading Cloudwatch events as newline-delimited JSON (ndjson) from stdin"
+      stdin_source(&block)
+    end
+  end
+
+  def stdin_source(&block)
+    $stdin.each_line do |line|
+      next if line.blank?
+      event = JSON.parse(line)
+      block.call(event)
```

Comment:
micro-optimization! don't assign a variable to the block argument, it saves an allocation on the heap, use the `yield` keyword which uses the stack version

Also added some docs to indicate that it takes a block

```suggestion
  # @yield calls block for each parsed event
  # @yieldparam [Hash] event
  def stdin_source
    $stdin.each_line do |line|
      next if line.blank?
      event = JSON.parse(line)
      yield event
```